Sources/GRPCReflectionService/Server/ReflectionService.swift

```diff
@@ -151,12 +165,21 @@ internal struct ReflectionServiceData: Sendable {
   }
 
   internal func nameOfFileContainingExtension(
-    named extendeeName: String,
+    extendeeName: String,
     fieldNumber number: Int32
   ) -> String? {
     let key = ExtensionDescriptor(extendeeTypeName: extendeeName, fieldNumber: number)
     return self.fileNameByExtensionDescriptor[key]
   }
+
+  // Returns nil if the type has no extensions or if it doesn't exist.
```

Comment:
This behaviour is surprising: `nil` usually means the thing we were looking for didn't exist (e.g. consider `dictionary["foo"]`). Having it also return `nil` when the type exists but doesn't have any extensions isn't a good API because it places the onus on the caller to work out whether the type exists or not.

In this case we should probably just do the work upfront and ensure that `fieldNumbersByType` contains the right information, i.e. empty arrays when the type exists but has no extensions.

Reply:
This is still not right: the only reason we store `messageTypeNames` is to check whether a message exists in this function, it isn't used anywhere else. 

When we create and populate `fieldNumbersByType` we should do so such it has a value for every type we know about, even if the value is an empty array. It means we do the work upfront and the semantics are clear: if a value exists then the type exists, if no value exists then we don't know about the type.